lib/Dialect/HW/HWOps.cpp

```diff
@@ -1733,6 +1734,10 @@ size_t InstanceOp::getPortIdForOutputId(size_t idx) {
 
 Value InstanceOp::getValue(size_t idx) {
   auto mpi = getPortList();
+  return getValue(idx, mpi);
+}
+
+Value InstanceOp::getValue(size_t idx, ModulePortInfo &mpi) {
```

Comment:
Does this fix the original regression? It looks it takes O(N^2) still (maybe I'm wrong).